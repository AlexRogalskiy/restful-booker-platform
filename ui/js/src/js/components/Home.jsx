import React from 'react';
import fetch from 'node-fetch';
import HotelRoomInfo from './HotelRoomInfo.jsx';
import HotelMap from './HotelMap.jsx';
import HotelLogo from './HotelLogo.jsx';
import HotelContact from './HotelContact.jsx';
import { API_ROOT } from '../api-config';

export default class Home extends React.Component {

    constructor() {
        super();

        this.state = {
            rooms : [{
                roomid: 1,
                roomNumber: 101,
                type: 'Standard Room',
                beds: 2,
                accessible: false,
                image : 'https://www.mwtestconsultancy.co.uk/img/room1.jpg',
                description: 'Aenean porttitor mauris sit amet lacinia molestie. In posuere accumsan aliquet. Maecenas sit amet nisl massa. Interdum et malesuada fames ac ante.',
                features: [
                    'Internet/Wi-fi',
                    'Jacuzzi Bathroom',
                    'Air conditioning',
                    'High Definition TV',
                    'Mini-bar'
                ]
            },{
                roomid: 2,
                roomNumber: 102,
                type: 'Single',
                beds: 1,
                accessible: true,
                image: 'https://www.mwtestconsultancy.co.uk/img/room2.jpg',
                description: 'Aenean porttitor mauris sit amet lacinia molestie. In posuere accumsan aliquet. Maecenas sit amet nisl massa. Interdum et malesuada fames ac ante.',
                features: [
                    'Internet/Wi-fi',
                    'Air conditioning',
                    'Mini-bar'
                ]
            }]
        }
    }

    componentDidMount() {
        fetch(API_ROOT.branding + '/branding/', {
			method: 'GET',
			headers: {
				'Accept': 'application/json',
				'Content-Type': 'application/json'
			}
        })
        .then(res => res.json())
        .then(res => {
            this.setState(res);
        })
        .catch(e => console.log(e));
    }

    render() {
        return (
            <div>
                <div className="container-fluid">
                    <div className="row">
                        <div className='col-sm-12 text-center'>
                            <HotelLogo logoDetails={this.state.logo} />
                        </div>
                    </div>
                    <div className="row hotel-description">
                        <div className='col-sm-1'></div>
                        <div className='col-sm-10'>
                            <p>{this.state.description}</p>
                        </div>
                        <div className='col-sm-1'></div>
                    </div>
                    <div className="row room-header">
                        <div className='col-sm-1'></div>
                        <div className='col-sm-10'>
                            <h2>Rooms</h2>
                        </div>
                        <div className='col-sm-1'></div>
                    </div>
                    {this.state.rooms.map((roomDetails) => {
                        return <div key={roomDetails.roomid}><HotelRoomInfo room={roomDetails} /></div>
                    })}
                        <HotelContact contact={this.state.contact} />
                    <div className="row">
                        <div className='col-sm-12'></div>
                        <HotelMap mapDetails={this.state.map} />
                    </div>
                </div>
            </div>
        );
    }
}
