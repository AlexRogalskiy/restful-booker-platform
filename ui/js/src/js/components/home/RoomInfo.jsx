import React from 'react';

export default class RoomInfo extends React.Component {

    constructor(){
        super();
    }

    render(){
        return(
            <div class="row">
                <div class="col-sm-12">
                    <p>{this.props.room.type}</p>
                    <p>{this.props.room.beds}</p>
                    <p>{this.props.room.accessible}</p>
                    <p>{this.props.room.details}</p>
                </div>
            </div>
        )
    }

}