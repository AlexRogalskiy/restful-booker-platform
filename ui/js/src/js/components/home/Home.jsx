import React from 'react';
import RoomInfo from './RoomInfo.jsx';
import HotelMap from './HotelMap.jsx';

export default class Home extends React.Component {

    constructor() {
        super();

        this.state = {
            rooms : [{
                roomid: 1,
                roomNumber: 101,
                type: "Twin",
                beds: 2,
                accessible: false,
                details: "Wifi, TV, Mini-bar"
            },{
                roomid: 2,
                roomNumber: 102,
                type: "Single",
                beds: 1,
                accessible: true,
                details: "Walk in shower"
            }],
            map : {
                hotelName : 'Shady meadows B&B',
                latitude : 52.6351204,
                longitude : 1.2733774
            }
        }
    }

	render() {
    	return(
			<div>
                <div className="row">
                    <div className='col-sm-4'></div>
                    <div className='col-sm-4'>
                        <h1>Hotel logo</h1>
                    </div>
                    <div className='col-sm-4'></div>
                </div>
                <div className="row">
                    <div className='col-sm-4'></div>
                    <div className='col-sm-4'>
                        <hr />
                        <h2>Hotel description</h2>
                    </div>
                    <div className='col-sm-4'></div>
                </div>
                <div className="row">
                    <div className='col-sm-4'></div>
                    <div className='col-sm-4'>
                        <hr />
                        <h2>Rooms listing</h2>
                    </div>
                    <div className='col-sm-4'></div>
                </div>
                {this.state.rooms.map((roomDetails) => {
					return <div key={roomDetails.roomid}><RoomInfo room={roomDetails} /></div>
				})}
                <div className="row">
                    <div className='col-sm-4'></div>
                    <div className='col-sm-4'>
                        <hr />
                        <h3>Location</h3>
                    </div>
                    <div className='col-sm-4'></div>
                </div>
                <div className="row">
                    <div className="col-sm-12">
                        <HotelMap mapDetails={this.state.map}/>
                    </div>
                </div>
                <div className="row">
                    <div className='col-sm-4'></div>
                    <div className='col-sm-4'>
                        <hr />
                        <h4>Footer</h4>
                    </div>
                    <div className='col-sm-4'></div>
                </div>
            </div>
      );
    }
}
