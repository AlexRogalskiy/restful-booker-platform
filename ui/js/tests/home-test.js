import React from 'react';
import Home from '../src/js/components/home/Home.jsx';
import RoomInfo from '../src/js/components/home/RoomInfo.jsx';
import HotelMap from '../src/js/components/home/HotelMap.jsx';

test('Home page renders', () => {
    const homeState = {
        rooms : [{
            roomid: 1,
            roomNumber: 101,
            type: "Twin",
            beds: 2,
            accessible: false,
            details: "Wifi, TV, Mini-bar"
        },{
            roomid: 2,
            roomNumber: 102,
            type: "Single",
            beds: 1,
            accessible: true,
            details: "Walk in shower"
        }],
        map : {
            hotelName : 'Shady meadows B&B',
            latitude : 52.6351204,
            longitude : 1.2733774
        }
    }

    const homeComponent = shallow(
        <Home />
    )

    homeComponent.setState(homeState);
    homeComponent.update();

    expect(homeComponent).toMatchSnapshot();
});

test('Room info for home page renders', () => {
    const roomDetails = {
        roomid: 1,
        roomNumber: 101,
        type: "Twin",
        beds: 2,
        accessible: false,
        details: "Wifi, TV, Mini-bar"
    }

    const roomInfoComponent = shallow(
        <RoomInfo room={roomDetails} />
    )

    expect(roomInfoComponent).toMatchSnapshot();
});

test('Map for home page renders', () => {
    const mapDetails = {
        hotelName : 'Shady Meadows',
        latitude : 52.6351204,
        longitude : 1.2733774
    }

    const mapComponent = shallow(
        <HotelMap mapDetails={mapDetails} />
    )

    expect(mapComponent).toMatchSnapshot();
});

test('Map for home page renders error', () => {
    const mapComponent = shallow(
        <HotelMap />
    )

    expect(mapComponent).toMatchSnapshot();
});