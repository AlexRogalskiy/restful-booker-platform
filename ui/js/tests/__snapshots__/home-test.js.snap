// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page renders 1`] = `
<div>
  <div
    className="row"
  >
    <div
      className="col-sm-4"
    />
    <div
      className="col-sm-4"
    >
      <h1>
        Hotel logo
      </h1>
    </div>
    <div
      className="col-sm-4"
    />
  </div>
  <div
    className="row"
  >
    <div
      className="col-sm-4"
    />
    <div
      className="col-sm-4"
    >
      <hr />
      <h2>
        Hotel description
      </h2>
    </div>
    <div
      className="col-sm-4"
    />
  </div>
  <div
    className="row"
  >
    <div
      className="col-sm-4"
    />
    <div
      className="col-sm-4"
    >
      <hr />
      <h2>
        Rooms listing
      </h2>
    </div>
    <div
      className="col-sm-4"
    />
  </div>
  <div
    key="1"
  >
    <RoomInfo
      room={
        Object {
          "accessible": false,
          "beds": 2,
          "details": "Wifi, TV, Mini-bar",
          "roomNumber": 101,
          "roomid": 1,
          "type": "Twin",
        }
      }
    />
  </div>
  <div
    key="2"
  >
    <RoomInfo
      room={
        Object {
          "accessible": true,
          "beds": 1,
          "details": "Walk in shower",
          "roomNumber": 102,
          "roomid": 2,
          "type": "Single",
        }
      }
    />
  </div>
  <div
    className="row"
  >
    <div
      className="col-sm-4"
    />
    <div
      className="col-sm-4"
    >
      <hr />
      <h3>
        Location
      </h3>
    </div>
    <div
      className="col-sm-4"
    />
  </div>
  <div
    className="row"
  >
    <div
      className="col-sm-12"
    >
      <HotelMap
        mapDetails={
          Object {
            "hotelName": "Shady meadows B&B",
            "latitude": 52.6351204,
            "longitude": 1.2733774,
          }
        }
      />
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-sm-4"
    />
    <div
      className="col-sm-4"
    >
      <hr />
      <h4>
        Footer
      </h4>
    </div>
    <div
      className="col-sm-4"
    />
  </div>
</div>
`;

exports[`Map for home page renders 1`] = `
<div
  className="map"
>
  <Map
    animate={true}
    animateMaxScreens={5}
    defaultCenter={
      Array [
        52.6351204,
        1.2733774,
      ]
    }
    defaultZoom={17}
    height={600}
    limitBounds="center"
    maxZoom={18}
    metaWheelZoom={false}
    metaWheelZoomWarning="Use META+wheel to zoom!"
    minZoom={1}
    mouseEvents={true}
    provider={[Function]}
    touchEvents={true}
    twoFingerDrag={false}
    twoFingerDragWarning="Use two fingers to move the map"
    warningZIndex={100}
    width={1024}
    zoomSnap={true}
  >
    <Marker
      anchor={
        Array [
          52.6351204,
          1.2733774,
        ]
      }
      key="marker_Shady Meadows"
      payload="Shady Meadows"
    />
  </Map>
</div>
`;

exports[`Map for home page renders error 1`] = `
<div
  className="map"
>
  <p>
    ERROR: Invalid or details missing for map to render
  </p>
</div>
`;

exports[`Room info for home page renders 1`] = `
<div
  class="row"
>
  <div
    class="col-sm-12"
  >
    <p>
      Twin
    </p>
    <p>
      2
    </p>
    <p />
    <p>
      Wifi, TV, Mini-bar
    </p>
  </div>
</div>
`;
